proposals/XXXX-debug.md

```diff
@@ -0,0 +1,119 @@
+# Introducing Configuration Tests to detect Debugging Conditions
+
+* Proposal: SE-TBD
+* Author(s): [Erica Sadun](http://github.com/erica), [John Holdsworth](https://github.com/johnno1962)
+* Review manager: TBD
+* Status: **Preliminary Implementation** ([preliminary diffs](https://github.com/apple/swift/compare/master...erica:optcheck))
+
+<!---
+* Implementation: [apple/swift#NNNNN](https://github.com/apple/swift/pull/NNNNN)
+* Decision Notes: [Rationale](https://lists.swift.org/pipermail/swift-evolution/), [Additional Commentary](https://lists.swift.org/pipermail/swift-evolution/)
+* Bugs: [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN), [SR-MMMM](https://bugs.swift.org/browse/SR-MMMM)
+* Previous Revision: [1](https://github.com/apple/swift-evolution/blob/...commit-ID.../proposals/NNNN-filename.md)
+* Previous Proposal: [SE-XXXX](XXXX-filename.md)
+* -->
+
+## Introduction
+
+This proposal introduces buildtime and runtime configuration tests to detect debugging conditions. This supports code exclusion at build time and code selection at run-time based on whether a project has been built for internal development or external release.
+
+This proposal was discussed on-list in the [Introducing a Debug Build Configuration Test](http://article.gmane.org/gmane.comp.lang.swift.evolution/12140) thread and in-forum in the [Support debug-only code](https://forums.swift.org/t/support-debug-only-code/11037/7) thread.
```

Comment:
The original Gmane link doesn't show the entire thread. Please could this be changed to a Swift Forums link?

```diff
-(http://article.gmane.org/gmane.comp.lang.swift.evolution/12140)
+(https://forums.swift.org/t/draft-introducing-a-debug-build-configuration-test/1794)
```

Reply:
Made suggested change.